atmo/jobs/jobs.py

```diff
@@ -13,9 +13,9 @@ def run_jobs():
     """
     run_jobs = []
     for job in SparkJob.objects.all():
-        # first let's update the status to update most_recent_status
-        job.update_status()
-        job.save()
+        # first let's update the status if there is a prior run
+        if job.latest_run:
```

Comment:
Going through all jobs and fetching all latest runs can be expensive the more jobs and runs we add. Can we find a different way to iterate over the job runs here?